src/cares_wrap.cc

```diff
@@ -1360,7 +1368,20 @@ class QueryAWrap: public QueryWrap {
   }
 
   int Send(const char* name) override {
-    AresQuery(name, ns_c_in, ns_t_a);
+    auto prop_value = GetField("search");
+    bool bSearch = false;
+
+    if (!prop_value.IsEmpty()) {
+      auto search_prop = prop_value.ToLocalChecked()->ToBoolean();
+      bSearch = search_prop->Value();
+    }
+
+    if (bSearch) {
+      AresSearch(name, ns_c_in, ns_t_a);
+    } else {
+      AresQuery(name, ns_c_in, ns_t_a);
+    }
+
     return 0;
   }
```

Comment:
Rather than looking up properties this is arguably better handled by making `QueryWrap` a template that takes a function pointer:
```c++
template <void (*F)(const char*, int, int)>
class BaseWrap {
  // ...
  void AresQuery(const char* name, int dnsclass, int type) {
    channel_->EnsureServers();
    F(channel_->cares_channel(), name, dnsclass, type, Callback,
      static_cast<void*>(this));
  }
  // ...
};

using QueryWrap = BaseWrap<ares_query>;
using SearchWrap = BaseWrap<ares_search>;
```
And then do the same to `QueryAWrap` and `QueryAaaaWrap`:
```c++
template <typename BaseT>
class BaseAWrap: public BaseT {
  // ... - no changes needed to body
};

using QueryAWrap = BaseAWrap<QueryWrap>;
using SearchAWrap = BaseAWrap<SearchWrap>;

// Repeat for QueryAaaaWrap
```

Reply:
No problem I'll make the changes.
A couple of Qs.

* What is the guideline when handling code on both C++ and JS, should I try to make most of the changes on the JS side?

* Where will I dispatch the correct object according to the search option? 
If it's in the lib side (Javascript) then I'll need to direct the resolve4 and resolve6 mapping  in the resolveMap (dns.js:240) to a new js function that will call the correct object, Is that correct?

Reply:
* How can I Test the feature?

Reply:
Re: resolveMap, that's correct.

Re: testing: I don't know.  It depends too much on the configuration of the system, I think.